pkg/kine/drivers/generic/generic.go

```diff
@@ -200,7 +225,7 @@ func openAndTest(driverName, dataSourceName string) (*sql.DB, error) {
 	return db, nil
 }
 
-func Open(ctx context.Context, driverName, dataSourceName string, paramCharacter string, numbered bool) (*Generic, error) {
+func Open(ctx context.Context, driverName, dataSourceName string, connPoolConfig ConnectionPoolConfig, paramCharacter string, numbered bool) (*Generic, error) {
```

Comment:
While it doesn't make a lot of difference here, in general options and configurations are passed as pointers in Go. 

Can we use a pointer here (and in all other usages of this struct?)